### PR TITLE
Rework support for static library archives

### PR DESCRIPTION
**Pull request by danielpieczko**
Here's the implementation for the improved static library support. Once reviewed, I figured it can be merged into your branch, and then we can have a pull request from your branch into xmos/xcommon_cmake develop so that Jenkins is run (which should pass - with these changes all the tests pass for me locally).

### CODE REVIEW

File: xcommon.cmake [hunk 1]
```diff
@@ -794,48 +826,63 @@ function(XMOS_REGISTER_DEPS)
...
-        target_include_directories(${LIB_NAME}-${lib_arch} PRIVATE ${LIB_INCLUDES})
+    foreach(archive ${LIB_ARCHIVES})
         if(NOT BUILD_NATIVE)
-            target_compile_options(${LIB_NAME}-${lib_arch} PUBLIC ${LIB_COMPILER_FLAGS} "-march=${lib_arch}")
+            if(LIB_ARCHIVE_ARCHS_${archive})
+                set(arch_list ${LIB_ARCHIVE_ARCHS_${archive}})
+            elseif(LIB_ARCHIVE_ARCHS)
+                set(arch_list ${LIB_ARCHIVE_ARCHS})
+            else()
+                set(arch_list xs3a)
```
Comment (xross) [inline xcommon.cmake]: Id be temped to have a DEFAULT_ARCH variable at the top somewhere (xs3a hardcoded a couple of times)

File: xcommon.cmake [hunk 2]
```diff
@@ -871,40 +918,60 @@ function(XMOS_REGISTER_STATIC_LIB)
         # Configure build of static lib
         XMOS_STATIC_LIBRARY()
     else()
-        # Configure use of build static lib
-        set(archive_name ${LIB_NAME})
-        if(NOT ${LIB_NAME} MATCHES "^lib")
-            set(archive_name "lib${LIB_NAME}")
+
+        # Handle any additional source files as though they are in a standard module repository
+        configure_lib(TRUE)
```
Comment (xross) [inline xcommon.cmake:926]: Nice code re-use.

File: xcommon.cmake (hunk 1 above)
Comment (danielpieczko) [reply xcommon.cmake]: Done - this makes it much clearer why it falls back through the if-else block.

**Comment by xross** [comment]
Great stuff :)

**Pull request merged**